src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -311,6 +312,15 @@
    jruby-request/wrap-with-error-handling
    ring/wrap-params))
 
+(schema/defn ^:always-validate valid-static-file-path?
+  "Helper function to decide if a static_file_content path is valid.
+  The access here is designed to mimic Puppet's file_content endpoint."
+  [path :- schema/Str]
+  (when-let [canonicalized-path (URIUtil/canonicalPath path)]
+    (or
+     (bidi.bidi/match-route [["modules/" :module-name "/files/" [#".*" :rest]] :helper] canonicalized-path)
+     (bidi.bidi/match-route [["environments/" :environment-name "/modules/" :module-name "/files/" [#".*" :rest]] :helper] canonicalized-path))))
```

Comment:
I was taking this from both the ticket and from the direct puppet google doc, which both indicate that environment may be in the path. The google doc says "<path> is relative to the master's codedir", which means that environments may be part of it.